backend/docs/multi_pathogen.md

```diff
@@ -0,0 +1,205 @@
+
+## Preprocessing
+
+TODO: Minimizer index, etc?
+
+## Preprocessing output
+
+```json
+{
+  "accession": "key1",
+  "version": 1,
+  "data": {
+    "metadata": {...},
+    "unalignedNucleotideSequences": {
+      "main": "ACTG"
+      // TODO: or do we also need multiple segments here?
+    },
+    "alignedNucleotideSequences": {
+      "matchingSuborganism": "ACTG",
+      "otherSuborganism": null
+    },
+    "nucleotideInsertions": {
+      "matchingSuborganism": [...],
+      "otherSuborganism": []
+    },
+    "alignedAminoAcidSequences": {
+      "matchingSuborganism_gene1": "NRNR*",
+      "matchingSuborganism_gene2": "MRGTNF*",
+      "otherSuborganism_gene1": null,
+      "otherSuborganism_gene2": null
+    },
+    "aminoAcidInsertions": {
+      "matchingSuborganism_gene1": [...],
+      "matchingSuborganism_gene2": [...],
+      "otherSuborganism_gene1": [],
+      "otherSuborganism_gene2": []
+    },
+    "files": {
+      "raw_reads": [
+        // anything relevant here?
+      ],
+      "sequencing_logs": []
+    }
+  },
+  "errors": [...],
+  "warnings": [...]
+}
+```
+
+similar for multiple segments.
+
+## LAPIS / SILO
+
+Will there be per-segment metadata?
+
+The reference genome will be a product "suborganism x segment":
+
+```json
+{
+  "nucleotideSequences": [
+    {"name": "suborganism1", "sequence": "..."},
+    {"name": "suborganism2", "sequence": "..."}
+  ],
+  "genes": [
+    {"name": "suborganism1_gene1", "sequence": "..."},
+    {"name": "suborganism1_gene2", "sequence": "..."},
+    {"name": "suborganism2_gene1", "sequence": "..."},
+    {"name": "suborganism2_gene2", "sequence": "..."}
+  ]
+}
+```
+
+for multi-segment:
+
+```json
+{
+  "nucleotideSequences": [
+    {"name": "suborganism1_segment1", "sequence": "..."},
+    {"name": "suborganism1_segment2", "sequence": "..."},
+    {"name": "suborganism2_segment1", "sequence": "..."},
+    {"name": "suborganism2_segment2", "sequence": "..."}
+  ],
+  "genes": [
+    {"name": "suborganism1_gene1", "sequence": "..."},
+    {"name": "suborganism1_gene2", "sequence": "..."},
+    {"name": "suborganism2_gene1", "sequence": "..."},
+    {"name": "suborganism2_gene2", "sequence": "..."}
+  ]
+}
+```
+
+## Website
+
+### Sequence entry details page
+
+Don't show sequences that are not there anyway.
+Only show the "segments" of the relevant suborganism.
+Only show the relevant genes.
+Strip the suborganism name for the sequence names.
+![img.png](sequenceDisplay.png)
+
+Do we need all the references here?
+![img_1.png](mutationReferences.png)
+
+Mutations should not show the suborganism name:
+![img_2.png](mutations.png)
+
+TODO: Is there more?
+
+### Search Page
+
+Mutation and lineage search should only show up once the search has been narrowed down to a specific suborganism.
+
+Mutation filter: Also strip the suborganism name for the user?
+User inputs `A123T`, we send `suborganism1:A123T` to LAPIS.
+(Similar for amino acid mutations).
```

Comment:
This also holds for insertions (if we do not make changes to LAPIS): the query to LAPIS would be `ins_<suborganism>:123:AATTGG` or `ins_<suborganism>-<gene>:123:LP`.